ce/ce/cli/project.ts

```diff
@@ -16,7 +16,17 @@ export interface ActivationOptions {
   json?: Uri;
 }
 
-export async function activate(session: Session, allowStacking: boolean, stackEntries: Array<string>, artifacts: Array<ResolvedArtifact>, registries: RegistryDisplayContext, options?: ActivationOptions) : Promise<boolean> {
+function trackActivationPlan(session: Session, resolved: Array<ResolvedArtifact>) {
+  for (let idx = 0; idx < resolved.length; ++idx) {
+    const artifact = resolved[idx].artifact;
+    if (artifact instanceof Artifact) {
+      session.trackActivate(artifact.registryUri.toString(), artifact.id, artifact.version);
+    }
+  }
+}
+
+export async function activate(session: Session, allowStacking: boolean, stackEntries: Array<string>, artifacts: Array<ResolvedArtifact>, registries: RegistryDisplayContext, options?: ActivationOptions): Promise<boolean> {
```

Comment:
To other reivewers: GitHub's diff is terrible. The interface of activate didn't change, it just had `trackActivationPlan(session, artifacts);` added to the first line.